android/modules/ui/src/java/ti/modules/titanium/ui/widget/picker/TiUIDatePicker.java

```diff
@@ -1,133 +1,210 @@
 /**
  * Appcelerator Titanium Mobile
- * Copyright (c) 2009-2012 by Appcelerator, Inc. All Rights Reserved.
+ * Copyright (c) 2009-2021 by Axway, Inc. All Rights Reserved.
  * Licensed under the terms of the Apache Public License
  * Please see the LICENSE included with this distribution for details.
  */
 package ti.modules.titanium.ui.widget.picker;
 
+import android.os.Build;
+import android.view.View;
+import android.widget.DatePicker;
+import android.widget.DatePicker.OnDateChangedListener;
+import android.widget.EditText;
+import androidx.annotation.NonNull;
+import com.google.android.material.textfield.TextInputLayout;
+import java.text.DateFormat;
 import java.util.Calendar;
 import java.util.Date;
-
+import java.util.HashMap;
 import org.appcelerator.kroll.KrollDict;
+import org.appcelerator.kroll.KrollFunction;
+import org.appcelerator.kroll.KrollObject;
 import org.appcelerator.kroll.KrollProxy;
 import org.appcelerator.kroll.common.Log;
+import org.appcelerator.titanium.R;
 import org.appcelerator.titanium.TiC;
-import org.appcelerator.titanium.proxy.TiViewProxy;
 import org.appcelerator.titanium.util.TiConvert;
-import org.appcelerator.titanium.util.TiRHelper;
-import org.appcelerator.titanium.util.TiRHelper.ResourceNotFoundException;
+import org.appcelerator.titanium.util.TiUIHelper;
 import org.appcelerator.titanium.view.TiUIView;
-
-import android.app.Activity;
-import android.widget.DatePicker;
-import android.widget.DatePicker.OnDateChangedListener;
+import ti.modules.titanium.ui.PickerProxy;
+import ti.modules.titanium.ui.UIModule;
 
 public class TiUIDatePicker extends TiUIView implements OnDateChangedListener
 {
-	private boolean suppressChangeEvent = false;
 	private static final String TAG = "TiUIDatePicker";
+	private TiUIDatePicker.DialogCallback dialogCallback;
+	private boolean suppressChangeEvent;
+	private Date minDate;
+	private Date maxDate;
 
-	protected Date minDate;
-	protected Date maxDate;
-	protected int minuteInterval;
-
-	public TiUIDatePicker(TiViewProxy proxy)
+	public TiUIDatePicker(@NonNull PickerProxy proxy)
 	{
 		super(proxy);
-	}
-	public TiUIDatePicker(final TiViewProxy proxy, Activity activity)
-	{
-		this(proxy);
 		Log.d(TAG, "Creating a date picker", Log.DEBUG_MODE);
 
-		// A bug where PickerCalendarDelegate does not send events to the
-		// listener on API Level 21 (Android 5.0) for TIMOB-19192
-		// https://code.google.com/p/android/issues/detail?id=147657
-		// Work around is to use spinner view instead of calendar view
-		int datePickerSpinner;
-		try {
-			datePickerSpinner = TiRHelper.getResource("layout.titanium_ui_date_picker_spinner");
-		} catch (ResourceNotFoundException e) {
-			if (Log.isDebugModeEnabled()) {
-				Log.e(TAG, "XML resources could not be found!!!");
+		// Determine if we should use a spinner, calendar view, or text field.
```

Comment:
I think the real issue is we have conflicting properties `useSpinner`, `nativeSpinner`, `calendarView`... and now the new `datePickerStyle` which is the enum style we both want. I keep the other booleans around in case the proxy fails to create the text field, which should never happen but the fallback is there.